Docs pm 2.9 dh (#143)
mulesoft/docs-partner-manager: modules/ROOT/pages/deploy-message-flows.adoc

> = Deploy and Test a Message Flow

Reviewer: here

Anypoint Partner Manager dynamically validates the message flow configuration elements as you complete them and displays a green checkmark next to each of the validated elements.

After Partner Manager validates the message flow configuration elements, you can deploy and test the message flow. It is best to deploy the message flow in a sandbox environment. After you verify that the message flow is successful, you can undeploy it from the sandbox and redeploy it to your production environment.

> == Deploy Message Flows

Reviewer: ```suggestion
== Deploy a Message Flow
```

To minimize the impact of the service interruption caused by deploying a message flow, perform message flow deployments outside of your regular business hours. This issue does not exist on CloudHub because CloudHub provides a zero-downtime update of the AS2 endpoint application. [*Vijay, is this applicable only for AS2 endpoints - send or receive?*]

To deploy a message flow, click the *Deploy* button.

> The message flow is deployed to Mule runtime engine, and you can do the following:

Reviewer: ```suggestion
The message flow is deployed to Mule runtime engine, enabling you to:
```

* Start receiving messages from your partner

> * Use the *Activity* page to track the processed messages.

Reviewer: GLOBALLY inspect lists for consistent use/nonuse of periods.

== Test a Deployed Message Flow

After you deploy the message flow, send test messages to verify that:

* You can receive messages and acknowledgments from your partners.
* The DataWeave map works.
* Your backend applications receive the transformed messages.

Test your message flow in your Partner Manager sandbox environment by sending a test file to the receiving endpoint:

> . In the navigation menu, select *Activity* and select the transmission to test.

Reviewer: ```suggestion
. In the navigation menu, select *Activity* and the transmission to test.
```

. Scan for an exclamation point (!) in any of the boxes next to the displayed flow stages.
+

> Exclamation points indicate that there is an issue with the transmission. Address any issues you find.

Reviewer: c/b
```suggestion
. Scan for an exclamation point (!) in any of the boxes next to the displayed flow stages and address any issues.
```
